src/components/molecules/Chatbox/Chatbox.tsx

```diff
@@ -24,66 +30,46 @@ export interface ChatboxProps {
   sendMessage: SendMesssage;
   sendThreadReply: SendChatReply;
   deleteMessage: DeleteMessage;
+  displayPoll?: boolean;
 }
 
 export const Chatbox: React.FC<ChatboxProps> = ({
   messages,
   sendMessage,
   sendThreadReply,
   deleteMessage,
+  displayPoll,
 }) => {
-  const [isSendingMessage, setMessageSending] = useState(false);
+  const { createPoll, voteInPoll } = useVenuePoll();
 
   const [selectedThread, setSelectedThread] = useState<
     WithId<MessageToDisplay>
   >();
 
   const closeThread = useCallback(() => setSelectedThread(undefined), []);
 
-  const hasChosenThread = selectedThread !== undefined;
-
-  // This logic dissallows users to spam into the chat. There should be a delay, between each message
-  useEffect(() => {
-    if (isSendingMessage) {
-      setTimeout(() => {
-        setMessageSending(false);
-      }, 500);
-    }
-  }, [isSendingMessage]);
-
-  const { register, handleSubmit, reset, watch } = useForm<{
-    message: string;
-  }>({
-    mode: "onSubmit",
-  });
-
-  const sendMessageToChat = handleSubmit(({ message }) => {
-    setMessageSending(true);
-    sendMessage(message);
-    reset();
-  });
-
-  const sendReplyToThread = handleSubmit(({ message }) => {
-    if (!selectedThread) return;
-
-    setMessageSending(true);
-    sendThreadReply({ replyText: message, threadId: selectedThread.id });
-    reset();
-  });
-
-  const chatValue = watch("message");
+  const [activeOption, setActiveOption] = useState<ChatOption>();
```

Comment:
This should probably use `ChatOptionType` (the `enum`) rather than `ChatOption` (a nested object)

Reply:
@0xdevalias @mike-lvov 
I suppose the `ChatOption` is a better way to manage the active option because it's just a plain object of the current option
```
{
  icon: IconDefinition;
  name: string;
}
```
What is the advantage of using `ChatOptionType` instead of `ChatOption`?

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1343/commits/3e8ec8791c5e715576d66645965ca8d7cb65a00a